backend/user-service/src/models/user.model.ts

```diff
@@ -0,0 +1,47 @@
+import { Schema, model } from 'mongoose'
+import { IUser } from '../types/IUser'
+import { Proficiency } from '../types/Proficiency'
+import { Role } from '../types/Role'
+
+const UserSchema = new Schema<IUser>({
+    username: {
+        type: String,
+        required: true,
+        unique: true,
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+    },
+    password: {
+        type: String,
+        required: true,
+    },
+    role: {
+        type: String,
+        enum: Object.values(Role),
+        required: true,
+    },
+    proficiency: {
+        type: String,
+        enum: Object.values(Proficiency),
+        required: false,
+    },
+    createdAt: {
+        type: Date,
+        required: false,
+    },
+    updatedAt: {
+        type: Date,
+        required: false,
+        default: null,
+    },
+    deletedAt: {
```

Comment:
what is the purpose of storing `deletedAt` here?

Reply:
mostly for soft deletes and logging

Reply:
okay got it, just need to handle a small edge case where the user tries to create an account with the same credentials (username, email) as before (which he deleted). (they could be rejected cuz the username/email still exists in the db)